pysparkle/pysparkle/data_quality/data_quality_utils.py

```diff
@@ -0,0 +1,145 @@
+"""
+Data_quality utility functions to set up validations
+"""
+from typing import List, Dict
+from pyspark.sql import DataFrame
+
+from great_expectations.core.batch import RuntimeBatchRequest
+from great_expectations.core.expectation_configuration import (
+    ExpectationConfiguration)
+from great_expectations.core.expectation_validation_result import ExpectationSuiteValidationResult
+from great_expectations.core.expectation_suite import ExpectationSuite
+from great_expectations.data_context import BaseDataContext
+from great_expectations.data_context.types.base import DataContextConfig, \
+    FilesystemStoreBackendDefaults
+
+
+def create_datasource_config(datasource_name: str) -> BaseDataContext:
+    """
+    Given a string containing the datasource name, this function generates a data context instance
+
+    Args:
+        datasource_name: Name of the datasource to be validated
```

Comment:
Can you explain some more what is data source?

Reply:
https://docs.greatexpectations.io/docs/terms/datasource/ So generally we'll likely use the raw data source or table name here, we're inferring the schema anyway but it needs to be unique for the validators. This is standard GX language as above but we basically use it to generate the other required names.

Reply:
Thanks, I think adding this URL to the arg description would be helpful, but if you think this is obvious, feel free to resolve.

Reply:
We're using datasource name everywhere now so I'd be putting the URL in alot of places, I think leave it for now and then we can add it if needed in future